imod/prepare/topsystem/default_allocation_methods.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
```

Comment:
FYI: changing the dataclass to pydantic dataclass, would be as easy as:

```suggestion
from pydantic.dataclasses import dataclass
```

You can make this change when PR #1054 is merged, probably not for this PR.